cogs/customreactions.py

```diff
@@ -0,0 +1,1111 @@
+                c = conn.cursor()
+                t = (prompt_message, response, author.id, delete, anywhere, dm,
+                     not is_moderator)
+                c.execute(
+                    'INSERT INTO CustomReactions(Prompt, Response, UserID, DeletePrompt, Anywhere, DM, '
+                    'Proposal) VALUES(?,?,?,?,?,?,?)', t)
+                conn.commit()
+                conn.close()
+                self.rebuild_reaction_list()
+                self.rebuild_proposal_list()
+
+                if is_moderator:
+                    title = "Custom reaction successfully added!"
+                else:
+                    title = "Custom reaction proposal successfully submitted!"
+                description = "-Prompt: {}\n-Response: {}".format(
+                    prompt_message, response)
+                if delete:
+                    description = "{}\n-Will delete the message that calls the reaction".format(
+                        description)
+                if anywhere:
+                    description = "{}\n-Will activate the custom reaction if the prompt is anywhere in a message"\
+                        .format(description)
+                if dm:
+                    description = "{}\n-Will react in the DMs of the user who calls the reaction instead of the "\
+                                  "channel".format(description)
+
+                embed = discord.Embed(title=title, description=description)
+                embed.set_footer(text="Added by {}.".format(author),
+                                 icon_url=author.avatar_url)
+                await message.edit(embed=embed)
+
+                return
+
+            if reaction_emoji == "⏹":
+                await leave(message)
+                return
+
+        async def listcustomreacts(message, proposals):
+            if proposals:
+                current_list = self.proposal_list
+            else:
+                current_list = self.reaction_list
+
+            if not current_list:
+                if proposals:
+                    title = 'There are currently no custom reaction proposals in this server'
+                else:
+                    title = 'There are currently no custom reactions in this server'
+                embed = discord.Embed(title=title, delete_after=60)
+                await message.edit(embed=embed)
+                return
+
+            index_list = [
+                '[{}]'.format(i + 1) for i in range(len(current_list))
+            ]
+
+            prompts_and_responses = [
+                'Prompt: {}\nResponse: {}'.format(reaction[1], reaction[2])
+                for reaction in current_list
+            ]
+
+            reaction_dict = {
+                "names": index_list,
+                "values": prompts_and_responses
+            }
+
+            embed = discord.Embed(title="Loading...")
+            await message.edit(embed=embed)
+
+            await message.add_reaction("⏪")
```

Comment:
this add_reaction code is duplicated somewhere else in the file, should be functionized

Reply:
Resolved in 7d16cb6